daemon/network.go

```diff
@@ -113,7 +114,12 @@ func (daemon *Daemon) CreateNetwork(name, driver string, ipam network.IPAM, opti
 	}
 
 	nwOptions = append(nwOptions, libnetwork.NetworkOptionIpam(ipam.Driver, "", v4Conf, v6Conf))
-	nwOptions = append(nwOptions, libnetwork.NetworkOptionDriverOpts(options))
+	nwOptions = append(nwOptions, libnetwork.NetworkOptionDriverOpts(driverOptions))
+	if len(networkOptions) > 0 {
+		if _, ok := networkOptions[netlabel.Internal]; ok {
+			nwOptions = append(nwOptions, libnetwork.NetworkOptionInternalNetwork())
```

Comment:
yes. we should differentiate `driver` option from libnetwork specific `network` option.
libnetwork core will pass on the driver option to the drivers transparently, while act on the `network` option.